src/init.cpp

```diff
@@ -1043,6 +1064,53 @@ bool AppInit2(ThreadHandlerPtr threads)
         g_banman->DumpBanlist();
     }, DUMP_BANS_INTERVAL * 1000);
 
+    /** If this is not TestNet we check for updates on startup and daily **/
+    /** We still add to the scheduler regardless of the users choice however the choice is respected when they opt out**/
+    if (!fTestNet)
+    {
+        int64_t UpdateCheckInterval = 24;
+
+        // Save some cycles and only so this area if the argument exists
+        if (mapArgs.count("-updatecheckinterval"))
+        {
+            try
+            {
+                UpdateCheckInterval = GetArg("-updatecheckinterval", 24);
+                // trivial: Don't allow checks less then 1 hour apart of update checks to prevent server DDoS (what is a good value)
+                if (UpdateCheckInterval < 1)
+                {
+                    LogPrintf("UpdateChecker: Update check interval too small of %" PRId64 "; Defaulting to 24 hour intervals", UpdateCheckInterval);
+
+                    UpdateCheckInterval = 24;
+                }
+            }
+
+            catch (const std::exception& ex)
+            {
+                // Tell them the exception and what they had put in place
+                LogPrintf("UpdateChecker: Exception occured while obtaining interval for update checks (ex: %s -updatecheckinterval=%s); Defaulting to 24 hour intervals", ex.what(), GetArgument("-updatecheckinterval", ""));
+
+                UpdateCheckInterval = 24;
+            }
+        }
+
+        scheduler.scheduleEvery([]{g_UpdateChecker->CheckForLatestUpdate();}, UpdateCheckInterval * 60 * 60 * 1000);
+
+        if (!fDisableUpdateCheck)
```

Comment:
with regards to this. CScheduler does not actually have a removal of a schedule task without clearing everything. Its intended to run services when forever just for that forever. so this allows it to be place in the service queue however if they have the update check disabled it'll just return in the update check function itself (thus the bool fDisableUpdateCheck. this allows them to enable and disable actively in qt options dialog without impacting queue as it is there regardless. also we want to allow the custom time to be set so this has it set in there otherwise we may just revert back to default which might be inconvenient to the users choice since they made a configuration option. Banman is in secheduler and probably better this way then reseting the scheduler everytime this bool is changed.